### PR TITLE
refactor(notification): reduce the configuration parameter code

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

ref https://github.com/ant-design/ant-design/pull/40718
### 💡 Background and solution

把 notification 获取配置部分的代码省略了一些。
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Reduce the configuration parameter code for notification         |
| 🇨🇳 Chinese | 减少 notification 配置参数代码          |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 83d8752</samp>

Improved the state management of the `message` and `notification` components by using context and refactoring. This makes the code more consistent, efficient, and readable.

### <samp>🤖 Generated by Copilot at 83d8752</samp>

*  Rename `initializeMessageConfig` to `getMessageConfig` and use it to initialize and update the state of `messageConfig` in `components/message/index.tsx` ([link](https://github.com/ant-design/ant-design/pull/44372/files?diff=unified&w=0#diff-afb9eb50b70d2e8a149079009b7e3150485edaa7d56dec15d54e6f64da530499L87-R87), [link](https://github.com/ant-design/ant-design/pull/44372/files?diff=unified&w=0#diff-afb9eb50b70d2e8a149079009b7e3150485edaa7d56dec15d54e6f64da530499L100-R100), [link](https://github.com/ant-design/ant-design/pull/44372/files?diff=unified&w=0#diff-afb9eb50b70d2e8a149079009b7e3150485edaa7d56dec15d54e6f64da530499L110-R110))
* Replace multiple state variables with a single `notificationConfig` object and add a new function `getNotificationConfig` to get the current configuration from the global context in `components/notification/index.tsx` ([link](https://github.com/ant-design/ant-design/pull/44372/files?diff=unified&w=0#diff-ec815250b7c2f7267ba9b8a93ea48315da09b7060dc31f38646dd7db8964a965L62-R79), [link](https://github.com/ant-design/ant-design/pull/44372/files?diff=unified&w=0#diff-ec815250b7c2f7267ba9b8a93ea48315da09b7060dc31f38646dd7db8964a965L84-R86))

### CODE REVIEW

**Comment by Yuiai01** [comment]
> 哇，少了 100 多 B。
> 
> ## size-limit report 📦
> Path	Size
> `./dist/antd.min.js`	388.13 KB (-124 B 🔽)
> `./dist/antd-with-locales.min.js`	447.12 KB (-117 B 🔽)

居然，没想到能少这么多

**Review by afc163 — APPROVED** [review]

File: components/notification/index.tsx [hunk 1]
```diff
@@ -59,43 +59,31 @@ interface GlobalHolderRef {
...
-      bottom: nextBottom,
-    } = getGlobalContext();
-
-    setPrefixCls(nextGlobalPrefixCls);
-    setContainer(nextGlobalContainer);
-    setMaxCount(nextGlobalMaxCount);
-    setRTL(nextGlobalRTL);
-    setTop(nextTop);
-    setBottom(nextBottom);
+    setNotificationConfig(getNotificationConfig);
```
Comment (afc163) [inline components/notification/index.tsx:86]: 这句话看起来比较奇怪 `setXxx(getXxx);`。

**Review by kiner-tang — APPROVED** [review]

File: components/notification/index.tsx (hunk 1 above)
Comment (Yuiai01) [reply components/notification/index.tsx:86]: 这么一看确实是有点怪怪的，那这里是要改成 initializeNotificationConfig 还是 getGlobalNotificationConfig 呢

File: components/notification/index.tsx [hunk 2]
```diff
@@ -59,43 +59,32 @@ interface GlobalHolderRef {
...
-
-  const [api, holder] = useInternalNotification({
-    prefixCls,
-    getContainer: () => container!,
-    maxCount,
-    rtl,
-    top,
-    bottom,
-  });
+  const initializeNotificationConfig: () => GlobalConfigProps = () => {
```
Comment (afc163) [inline components/notification/index.tsx:62]:
```suggestion
  const initializeNotificationConfig: (): GlobalConfigProps => {
```

Comment (Yuiai01) [reply components/notification/index.tsx:62]: 这就去改

File: components/notification/index.tsx [hunk 3]
```diff
@@ -59,43 +60,32 @@ interface GlobalHolderRef {
...
-      bottom: nextBottom,
-    } = getGlobalContext();
-
-    setPrefixCls(nextGlobalPrefixCls);
-    setContainer(nextGlobalContainer);
-    setMaxCount(nextGlobalMaxCount);
-    setRTL(nextGlobalRTL);
-    setTop(nextTop);
-    setBottom(nextBottom);
+    setNotificationConfig(initializeNotificationConfig);
```
Comment (afc163) [inline components/notification/index.tsx:88]: 原来 setXxx 能接收一个 function 作为 setter，学习了。

File: components/notification/index.tsx [hunk 4]
```diff
@@ -59,43 +60,32 @@ interface GlobalHolderRef {
...
-  const [api, holder] = useInternalNotification({
-    prefixCls,
-    getContainer: () => container!,
-    maxCount,
-    rtl,
-    top,
-    bottom,
-  });
+  const initializeNotificationConfig = (): GlobalConfigProps => {
+    const { prefixCls, container, maxCount, rtl, top, bottom } = getGlobalContext();
```
Comment (afc163) [inline components/notification/index.tsx:64]: getGlobalContext 这里反复解构和 render，是不是封装一个这样的写法更省空间。

我不是很确定，要不要试试效果？

```
return getGlobalContext(['prefixCls', 'container', 'maxCount', 'rtl', 'top', 'bottom'])
```

Comment (Yuiai01) [reply components/notification/index.tsx:64]: 好，我晚点回家试试

Comment (Yuiai01) [reply components/notification/index.tsx:64]: 偏右大佬，我发现可以直接用上面的 getGlobalContext（测试用例也跑过了），不过需要稍稍改造一下，如下：
image
image

测试用例结果：
image

**Pull request merged**